src/stdlib/parse_influxdb.rs

```diff
@@ -0,0 +1,431 @@
+use std::collections::BTreeMap;
+
+use chrono::DateTime;
+use influxdb_line_protocol::{FieldValue, ParsedLine};
+
+use crate::compiler::prelude::*;
+use crate::{btreemap, value};
+
+fn influxdb_line_to_metrics(line: ParsedLine) -> Result<Vec<ObjectMap>, ExpressionError> {
+    let ParsedLine {
+        series,
+        field_set,
+        timestamp,
+    } = line;
+
+    let timestamp = timestamp.map(DateTime::from_timestamp_nanos);
+
+    let tags: Option<ObjectMap> = series.tag_set.as_ref().map(|tags| {
+        tags.iter()
+            .map(|t| (t.0.to_string().into(), t.1.to_string().into()))
+            .collect()
+    });
+
+    field_set
+        .into_iter()
+        .map(|f| {
+            let mut metric = ObjectMap::new();
+            let measurement = &series.measurement;
+            let field_key = f.0.to_string();
+            let field_value = match f.1 {
+                FieldValue::I64(v) => v as f64,
+                FieldValue::U64(v) => v as f64,
+                FieldValue::F64(v) => v,
+                FieldValue::Boolean(v) => {
+                    if v {
+                        1.0
+                    } else {
+                        0.0
+                    }
+                }
+                FieldValue::String(_) => {
+                    return Err(Error::StringFieldSetValuesNotSupported.into());
+                }
+            };
+
+            // `influxdb_line_protocol` crate seems to not allow NaN float values while parsing
+            // field values and this case should not happen, but just in case, we should
+            // handle it.
+            let Ok(field_value) = NotNan::new(field_value) else {
+                return Err(Error::NaNFieldSetValuesNotSupported.into());
+            };
+
+            let metric_name = format!("{measurement}_{field_key}");
+            metric.insert("name".into(), metric_name.into());
+
+            if let Some(tags) = tags.as_ref() {
+                metric.insert("tags".into(), tags.clone().into());
+            }
+
+            if let Some(timestamp) = timestamp {
+                metric.insert("timestamp".into(), timestamp.into());
+            }
+
+            metric.insert("kind".into(), "absolute".into());
+
+            let gauge_object = value!({
+                value: field_value
+            });
+            metric.insert("gauge".into(), gauge_object);
+
+            Ok(metric)
+        })
+        .collect()
+}
+
+#[derive(Debug, Clone, thiserror::Error)]
+enum Error {
```

Comment:
Created this error enum for convenience instead of working with raw strings. It is only used in Lines 42 and 50